lib/pbench/test/unit/server/test_upload.py

```diff
@@ -298,22 +286,59 @@ def test_empty_upload(
         self.verify_logs(caplog)
         assert not self.cachemanager_created
 
+    def test_temp_exists(
+        self, monkeypatch, client, tmp_path, server_config, pbench_drb_token
+    ):
+        md5 = "d41d8cd98f00b204e9800998ecf8427e"
+
+        def td_exists(self, *args, **kwargs):
+            """Mock out Path.mkdir()
+
+            The trick here is that calling the UPLOAD API results in two calls
+            to Path.mkdir: one in the __init__ to be sure that ARCHIVE/UPLOAD
+            exists, and the second for the temporary subdirectory. We want the
+            first to succeed normally so we'll pass the call to the real mkdir
+            if the path doesn't end with our MD5 value.
+            """
+            if self.name != md5:
+                return self.real_mkdir(*args, **kwargs)
+            raise FileExistsError(str(self))
+
+        filename = "tmp.tar.xz"
+        datafile = tmp_path / filename
+        datafile.write_text("compressed tar ball")
```

Comment:
This would have been better as a `BytesIO` instance.

Reply:
FYI, this is a straight `git cherry-pick` of the original commit.